Neos.FluidAdaptor/Classes/Core/Widget/AbstractWidgetViewHelper.php

```diff
@@ -238,19 +239,15 @@ protected function initiateSubRequest()
 
                 /** @var $parentResponse ActionResponse */
                 $parentResponse = $this->controllerContext->getResponse();
-                $parentResponse->setContent($subResponse->getContent());
-                $parentResponse->setStatusCode($subResponse->getStatusCode());
-                try {
-                    // TODO: With next major this part should just apply the sub (action) response to the parent (action) response
-                    $redirectUri = new Uri($subResponse->getHeader('Location'));
-                    $parentResponse->setRedirectUri($redirectUri, $subResponse->getStatusCode());
-                } catch (\InvalidArgumentException $innerException) {
-                    // FIXME: What do we do in this case? (probably nothing because there might not have been a location header).
-                }
+                $intoParentResponse = new IntoActionResponse($parentResponse);
+                $parentResponse = $subResponse->prepareRendering($intoParentResponse)->render();
 
                 throw $exception;
             }
         }
+
+        // At this point the sub reponse is already merged to the parent response,
+        // so it's slightly odd to return it here, but for now that's what we do.
```

Comment:
See above, we should return `$subResponse->getContent()` here, or rather through the `RendererInterface` API `return $subResponse->prepareRender(new Content())->render();`

Reply:
Yep, yep, I kind of agree, see longer comment below.